Wrappers/Python/cil/optimisation/algorithms/SPDHG.py

```diff
@@ -41,203 +45,422 @@ class SPDHG(Algorithm):
         A convex function with a "simple" proximal
     operator : BlockOperator
         BlockOperator must contain Linear Operators
-    tau : positive float, optional, default=None
-        Step size parameter for Primal problem
-    sigma : list of positive float, optional, default=None
-        List of Step size parameters for Dual problem
-    initial : DataContainer, optional, default=None
+    tau : positive float, optional, default= see note
+        Step size parameter for primal problem
+    sigma : list of positive float, optional, default= see note
+        List of Step size parameters for dual problem
+    initial : DataContainer, optional, default to a zero DataContainer in the range of the `operator`.
         Initial point for the SPDHG algorithm
-    prob : list of floats, optional, default=None
-        List of probabilities. If None each subset will have probability = 1/number of subsets
-    gamma : float
-        parameter controlling the trade-off between the primal and dual step sizes
+    gamma : float, optional
+            Parameter controlling the trade-off between the primal and dual step sizes
+    sampler: optional, an instance of a `cil.optimisation.utilities.Sampler` class or another class with the function __next__(self) implemented outputting an integer from {1,...,len(operator)}. 
+            Method of selecting the next index for the SPDHG update. If None, a sampler will be created for random sampling with replacement and each index will have probability = 1/len(operator)
+    prob_weights: optional, list of floats of length num_indices that sum to 1. Defaults to [1/len(operator)]*len(operator)
+            Consider that the sampler is called a large number of times this argument holds the expected number of times each index would be called,  normalised to 1. Note that this should not be passed if the provided sampler has it as an attribute. 
```

Comment:
Can we explain "Note that this should not be passed if the provided sampler has it as an attribute."?